pages/[wallet]/index.tsx

```diff
@@ -927,85 +931,108 @@ const RemoveContract: React.FC<IRemoveContract> = ({
       onClose={onClose}
     >
       <PopoverAnchor>
-        <Tooltip
-          p={0}
-          bg="transparent"
-          boxShadow="none"
-          label={
-            <Card py={2} px={4}>
-              <Text size="label.sm">
-                Remove this contract from the dashboard
-              </Text>
-            </Card>
-          }
+        <Button
+          padding={0}
+          ml="16px"
+          borderRadius="md"
+          variant="outline"
+          size="sm"
+          isLoading={isLoading}
+          onClick={(e) => {
+            e.stopPropagation();
+            e.preventDefault();
+            if (!isOpen) {
+              onOpen();
+            }
+          }}
         >
-          <Button
-            padding={0}
-            ml="16px"
-            borderRadius="md"
-            variant="outline"
-            size="sm"
-            isLoading={isLoading}
-            onClick={(e) => {
-              e.stopPropagation();
-              e.preventDefault();
-              if (chainId === targetChainId) {
-                mutate({
-                  contractAddress,
-                  contractType,
-                  chainId: targetChainId,
-                });
-              } else {
-                onOpen();
-              }
-            }}
-          >
-            <Icon as={FaTrash} boxSize={3} />
-          </Button>
-        </Tooltip>
+          <Icon as={FaTrash} boxSize={3} />
+        </Button>
       </PopoverAnchor>
 
       <PopoverContent>
-        <PopoverBody>
-          <Flex direction="column" gap={4}>
-            <Heading size="label.lg">
-              <Flex gap={2} align="center">
-                <Icon boxSize={6} as={AiOutlineWarning} />
-                <span>Network Mismatch</span>
-              </Flex>
-            </Heading>
+        {chainId === targetChainId ? (
+          <PopoverBody>
```

Comment:
to get the right background here looks like you need to do what the mismatchButton does:

```
<Card
          maxW="sm"
          w="auto"
          as={PopoverContent}
          bg="backgroundCardHighlight"
          mx={6}
          boxShadow="0px 0px 2px 0px var(--popper-arrow-shadow-color)"
        >
        ```